src/Orleans/Messaging/Message.cs

```diff
@@ -278,13 +280,27 @@ public ResponseTypes Result
             set {  Headers.Result = value; }
         }
 
-        public DateTime? Expiration
+        public TimeSpan? Expiration
         {
-            get { return Headers.Expiration; }
+            get
+            {
+                var elapsed = DateTime.UtcNow - localCreationTime;
+                Headers.Expiration = Headers.Expiration - elapsed;
+                return Headers.Expiration;
```

Comment:
If I access this property 100 times in succession, the header will expire much more quickly than it should, or am I misunderstanding the code?